lib/source/consul.js

```diff
@@ -7,6 +7,71 @@ const consul = require('consul');
 const DEFAULT_AGENT_ADDRESS = '127.0.0.1';
 const DEFAULT_AGENT_PORT = 8500;
 
+class ConquesoCompatHealthParser {
+  constructor() {
+    this.properties = {};
+  }
+
+  update(data, cluster) {
+    const properties = {};
+    const addresses = [];
+
+    data.forEach((info) => {
+      // Prefer the service address, not the Consul agent address.
+      if (info.Service && info.Service.Address) {
+        addresses.push(info.Service.Address);
+      } else if (info.Node && info.Node.Address) {
+        addresses.push(info.Node.Address);
+      }
+    });
+
+    properties.addresses = addresses.sort();
+    properties.cluster = cluster;
+
+    this.properties = properties;
+  }
+}
+
+class ConquesoCompatServiceParser {
+  constructor() {
+    this.properties = {};
+  }
+
+  update(data, callback) {
+    const services = Object.keys(data);
+    const watcherInfo = [];
+
+    services.forEach((service) => {
+      const tags = data[service];
```

Comment:
This doesn't solve for a tagging scheme of key-value pairs like `["version:1.2.3", "cluster:foo"]`. We need to check if there's a `key:value` format, then only pick `cluster:...`, or if there are no tags matching that format, assume that all tags are cluster names.